src/odemis/gui/comp/legend.py

```diff
@@ -365,3 +364,101 @@ def on_paint(self, _):
             self._max_tick_width = max_width
             self.SetMinSize((self._max_tick_width + 14, -1))
             self.Parent.GetSizer().Layout()
+
+
+class RadioLegend(wx.Panel):
+    """
+    This class describes a legend containing radio buttons,
+    where each button represents a possible selection of a VA.
+    """
+
+    def __init__(self, parent, wid=-1, pos=(0, 0), size=wx.DefaultSize, style=wx.NO_BORDER):
+        wx.Panel.__init__(self, parent, wid, pos, size, style)
+
+        # get colors: if ar-viewport is selected, to avoid colors for bg and fg for txt will be taken from selection
+        self.bg_color = parent.GetBackgroundColour()
+        self.fg_color = parent.GetForegroundColour()
+
+        self.SetBackgroundColour(self.bg_color)
+        self.SetForegroundColour(self.fg_color)
+
+        # descriptive VA text
+        self.text = wx.TextCtrl(self, value="Polarization", style=wx.NO_BORDER | wx.CB_READONLY)
+        self.text.SetBackgroundColour(self.bg_color)
+        self.text.SetForegroundColour(self.fg_color)
+        self.text.SetToolTipString("Polarization direction currently displayed")
+
+        # current polarization displayed in the legend (or None)
+        # can be radio buttons or text only depending on the number of polarization directions
+        self.pol_display = None
+        self.control_sizer = wx.BoxSizer(wx.HORIZONTAL)
+
+        # border_sizer is needed to add a border around the legend
+        self.border_sizer = wx.BoxSizer(wx.VERTICAL)
+        self.border_sizer.Add(self.control_sizer, border=8, flag=wx.ALL | wx.EXPAND)
+
+        self.SetSizer(self.border_sizer)
+
+    def clear(self):
+        """clear current legend"""
+        # when fct called, clear control_sizer and border_sizer to add new widgets depending on data
```

Comment:
Comments can be improved. Maybe the two comments can be merged into something like `Remove the display widget, to not show anything. It will be added again next time the polarization entries are set.`